syntax = "proto3";
package api;

message Welcome {
    string version = 1;
}

message Hello {
    string version = 1;
}

message StartOnvifDiscovery {
    int32 timeout = 1;
}

message OnvifDiscoveryResult {
    repeated string device = 1;
}

message LoadOnvifCameraSettings {
    string device = 1;
    string login = 2;
    string password = 3;
}

message OnvifCameraProfile {
    string name = 1;
    string encoding = 2;
    int32 width = 3;
    int32 height = 4;
    float framerate = 5;
    float bitrate = 6;
    string url = 7;
}

message OnvifCameraSettingsResult {
    string error = 1;
    string device = 2;
    string make = 3;
    string model = 4;
    repeated OnvifCameraProfile profile = 5;
}
